lambda/docker-runner.yaml

```diff
@@ -0,0 +1,202 @@
+# Based off of https://github.com/aws-samples/aws-lambda-efs-samples
+AWSTemplateFormatVersion: '2010-09-09'
+Transform: AWS::Serverless-2016-10-31
+Description: Lambda to run a docker container
+Metadata:
+  AWS::CloudFormation::Interface:
+    ParameterGroups:
+      - Label:
+          default: Container Image
+        Parameters:
+          - ImageUri
+          - EntryPoints
+          - Command
+          - WorkingDirectory
+          - MemorySize
+      - Label:
+          default: Synapse
+        Parameters:
+          - SynapsePAT
+      - Label:
+          default: Execution Trigger
+        Parameters:
+          - EnableSchedule
+          - Schedule
+      - Label:
+          default: EFS
+        Parameters:
+          - PerformanceMode
+          - Encrypted
+          - EfsPath
+Parameters:
+  PerformanceMode:
+    Type: String
+    Default: "generalPurpose"
+    Description: The EFS disk performance mode
+    AllowedValues:
+      - generalPurpose
+      - maxIO
+  Encrypted:
+    Type: String
+    Default: true
+    Description: True to enable EFS encryption, false to disable.
+    AllowedValues:
+      - true
+      - false
+  EfsPath:
+    Type: String
+    Description: Path to access EFS file store.
+    Default: /mnt/efs
+  MemorySize:
+    Type: Number
+    MinValue: 1024
+    MaxValue: 10240
+    Default: 1024
+    Description: >
+      The amount of memory (in MB) available to the function at runtime.
+      Increasing the function's memory also increases its CPU allocation.
+  ImageUri:
+    Type: String
+    Description: >
+      The URI of the Amazon ECR repository for the Lambda function's container image.
+      Example: 111111111.dkr.ecr.us-east-1.amazonaws.com/MY-DOCKER:latest
+  EntryPoints:
+    Type: CommaDelimitedList
+    Default: ""
+    Description: (Optional) Comma separated list of entry points to docker container
+  Command:
+    Type: CommaDelimitedList
+    Default: ""
+    Description: (Optional) Comma separated list of parameters to pass in with docker ENTRYPOINT
+  WorkingDirectory:
+    Type: String
+    Default: ""
+    Description: (Optional) The working directory for the docker container
+  EnableSchedule:
+    Description: >
+      true to run on a schedule, false to disable. If enabled a valid Schedule must be provided
+    Type: String
+    Default: false
+    AllowedValues:
+      - true
+      - false
+  Schedule:
+    Description: >
+      Schedule to execute the docker, can be a rate or a cron schedule. Format at
+      https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/ScheduledEvents.html
+    Type: String
+    Default: cron(0 09 ? * MON *)  # Run every Monday at 9am
+    ConstraintDescription: "Use schedule format: https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/ScheduledEvents.html"
+  SynapsePAT:
```

Comment:
This will store a Synapse secret in plaintext as a parameter to a cfn stack. The parameters can be read by the admin role in any scipool account, but not by communitymanagers or endusers (they can see the plaintext of their own stack parameters if they know the full stack id). I think this is still excessive, since a PAT is long lived and should be readable only to the Synapse account holder. `secretsmanager` and limits to IAM roles are probably the way to do it properly...

Reply:
secrets will appear as `****` in CFN parameters and secrets are not outputted so it will never appear in CFN outputs section.

Reply:
by adding `KmsKeyArn` to this template and defining the kms key in PR https://github.com/Sage-Bionetworks-IT/organizations-infra/pull/129  the env vars are no longer visible to any logged in user... https://pasteboard.co/JW6CNXg.png

Reply:
What do you think, @ahayden ?